swirl_lm/boundary_condition/immersed_boundary_method.py

```diff
@@ -317,6 +328,507 @@ def _check_mesh_size(a, b):
   return init_fn
 
 
+def init_ghost_point_helper_vars(
+    params: parameters_lib.SwirlLMParameters,
+    surface: tf.Tensor,
+    ib_interior_mask: FlowFieldVal,
+    dim: int,
+    my_var: str,
+    n_cores: initializer.ThreeIntTuple,
+) -> InitFn:
```

Comment:
Let's return the result dictionary, e.g. {'a': tf.Tensor(...), 'b': tf.Tensor(...), ...} directly from this function instead of an `init_fn` for each of these helper variables. We can reuse variables we computed earlier inside this function to prevent repeated operations. We'll need to call `partial_mesh_for_core` for each variable that depends on the coordinates inside this function.